### ISSUE TITLE
Some questions about GetImage() vs SkiaSharp.SKImage

### ISSUE THREAD

**Issue by pkq520** [ScottPlot/ScottPlot#4562]
**Question1:** Hi all, I'm using GetImage() right now to return what was plotted on the chart, but I found that it has two parameters, width and height, I've tried to change two values, and found that things stored in the change are blurry, and when it's slightly larger, it's better, but I don't know how these two parameters affect the image taken from the chart, is it simple to force scaling, and if I just want to get the optimal pixel display, can I write it like this？

```cs
double[] dataX = { 1, 2, 3, 4, 5 };
double[] dataY = { 1, 4, 9, 16, 25 };
AvaPlot1.Plot.Add.Scatter(dataX, dataY);
AvaPlot1.Refresh();
ScottPlot.Image scottImage = AvaPlot1.Plot.GetImage((int)AvaPlot1.Width, (int)AvaPlot1.Height);
var scottImage_byte =  scottImage.GetImageBytes();
scottImage.SavePng("./ScottPlot.png");
```
**Question2:** Another problem is that when I call ScottPlot.Image, I see that it is annotated as 'Bitmap representation of a SkiaSharp.SKImage', but I can't find a way to convert ScottPlot.Image to SkiaSharp.SKImage. (I'm sorry, there may be a problem with the issue itself, as I don't know much about SkiaSharp.SKImage and it's not clear if it's also a description class for images).

**ScottPlot Version:** 5.0.45

**Comment by swharden** [comment]
Hi @pkq520, thanks for this question!

I am on mobile right now so I will be brief, but can you tell me a little bit about what your goal is? Do you have an existing SKSurface you are trying to show a plot on? If so, note that Plot.Render() can render onto an existing SKSurface and these issues may go away 

If you are doing something different, I would be curious to learn what you are trying to achieve by getting the bitmap bites! I am very interested in your application and would love to help you realize your goals 🚀

**Comment by pkq520** [comment]
Thanks for the reply, what I'm talking about is actually two questions, although it may seem like one. The origin of these two problems is that I am going to export the diagram I have drawn and store it in a pdf. Since the work I'm doing requires dual-framework compatibility between .NET Framework 4.8 (using ScottPlot 4.0) and .net 8.0 (avalonia, using ScottPlot5.0), I wrote it this in NET Framework 4.8:
```cs
var Screen_Shot = new Bitmap(FormsPlot_Basic_Spectral.Width, FormsPlot_Basic_Spectral.Height);
FormsPlot_Basic_Spectral.DrawToBitmap(Screen_Shot, new Rectangle(0, 0, FormsPlot_Basic_Spectral.Width, FormsPlot_Basic_Spectral.Height));
```
Functions that are converted to byte arrays are also written:
```cs
public static byte[] ConvertBitmapToByteArray(Bitmap bitmap)
        {
            using (var stream = new MemoryStream())
            {
                bitmap.Save(stream, System.Drawing.Imaging.ImageFormat.Png); 
                return stream.ToArray();
            }
        }
```
I wrote it this in NET 8.0:
```cs
ScottPlot.Image scottImage = AvaPlot1.Plot.GetImage((int)AvaPlot1.Width, (int)AvaPlot1.Height);
var scottImage_byte =  scottImage.GetImageBytes();
```
Since I need to write pdf, I need to determine how to set it up to make the exported image have the best pixel effect, and because it needs to be converted to byte array, I want to know about the SkiaSharp.SKImage used by ScottPlot.Image, for example, the image has parameters similar to RGB ARGB, but there are no relevant parameters in the GetImageBytes function. and whether the results can be similar to the one achieved by the conversion function in net framework 4.8.
Another point is that the way I wrote in the .NET Framework 4.8 code to get the diagram may not be very good, please let me know if you have any suggestions.

**Comment by swharden** [comment]
Hi @pkq520, thanks for the additional detail!

I haven't been able to take a closer look yet, but perhaps reviewing the implementation of `GetBitmapBytes()` will help you do something similar manually

https://github.com/ScottPlot/ScottPlot/blob/2389ab65784525b67d4e23cb3cf8fabe3df5c0a5/src/ScottPlot5/ScottPlot5/Primitives/Image.cs#L142-L166

**Comment by swharden** [comment]
This bit may also be useful for you to review:

https://github.com/ScottPlot/ScottPlot/blob/2389ab65784525b67d4e23cb3cf8fabe3df5c0a5/src/ScottPlot5/ScottPlot5/Plot.cs#L265-L270

The `SKSurface` may now be saved or converted to a byte array in memory, bypassing `ScottPlot.Image`

**Comment by pkq520** [comment]
I'm sorry, I've been quite busy lately, I've read your reply and learned something, I've solved my problem. In these two days of testing, I also found that ScottPlot 5.0 will automatically re-render the chart according to the length and width set when acquiring the image, and the phenomenon is that the number of labels on the horizontal and vertical axis will change dynamically, rather than simply taking a screenshot of the existing chart, which is amazing. Thank you again for your help. 🚀
